File: src/org/elixir_lang/injection/ElixirSigilInjectionSupport.java
package org.elixir_lang.injection;

import com.intellij.psi.PsiLanguageInjectionHost;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.intellij.plugins.intelliLang.inject.AbstractLanguageInjectionSupport;

public final class ElixirSigilInjectionSupport extends AbstractLanguageInjectionSupport {
    @NonNls public static final String ELIXIR_SUPPORT_ID = "elixir";

    @Override
    @NotNull
    public String getId() {
        return ELIXIR_SUPPORT_ID;
    }

    @Override
    public Class<?> @NotNull [] getPatternClasses() {
        return new Class[] { ElixirSigilPatterns.class };
    }

    @Override
    public boolean isApplicableTo(com.intellij.psi.PsiLanguageInjectionHost host) {
        return true;
    }

    @Override
    public boolean useDefaultInjector(final PsiLanguageInjectionHost host) {
        return true;
    }
}

File: src/org/elixir_lang/injection/ElixirSigilInjectionSupport.kt
package org.elixir_lang.injection

import com.intellij.psi.PsiLanguageInjectionHost
import org.intellij.plugins.intelliLang.inject.AbstractLanguageInjectionSupport
import org.jetbrains.annotations.NonNls

internal class ElixirSigilInjectionSupport : AbstractLanguageInjectionSupport() {
    override fun getId(): String {
        return ELIXIR_SUPPORT_ID
    }

    override fun getPatternClasses(): Array<Class<*>> {
        return arrayOf(ElixirSigilPatterns::class.java)
    }

    override fun isApplicableTo(host: PsiLanguageInjectionHost): Boolean {
        return true
    }

    override fun useDefaultInjector(host: PsiLanguageInjectionHost): Boolean {
        return true
    }

    companion object {
        const val ELIXIR_SUPPORT_ID: @NonNls String = "elixir"
    }
}
